### ISSUE TITLE
Changing convention to build/runtime images

### ISSUE THREAD

**Issue by lumjjb** [nabla-containers/nabla-base-build#11]
Due to differences in the way applications are built, we should have some mechanism that separates the files required by the build from the runtime. 

I propose there be two types of base images, a `build` image and a `runtime` image. 

One example of how they would be used is an example of a multi-stage build process:

```
FROM nabla-go-build

COPY src/ /goapp/
WORKDIR "/goapp"

RUN go get -d ./...
RUN make -f Makefile.goapp

FROM nabla-go-runtime
COPY --from=0 /goapp/goapp.nabla /goapp.nabla
ENTRYPOINT ["/goapp.nabla"]
```

**Comment by djwillia** [comment]
Does this differ from how multi-stage builds are done with non-nabla containers?  Is there any convention there that we should be aware of?

**Comment by lumjjb** [comment]
Nope! this does not differ from how things are done with normal containers.

The issue here is that we always referred to the base as the bottom image (runtime), but with the introduction of golang and C in nabla, the base images in the `base-build` repo are actually the build images. This would be more of a naming consistency than actual functionality changes.

**Comment by ricarkol** [comment]
Wanted to illustrate what Brandon described above.

![img_20180926_104231 1](https://user-images.githubusercontent.com/11448949/46088535-97b20f80-c17a-11e8-92f3-4a4228f7f327.jpg)

We have a hello world app in go and js. The go one is built using the rumprun version of `go build` (the `go get` below) and then copied to the final image. The javascript one, `hello.js`, is just copied.

```
FROM nabla-go-base		    FROM node:4.3.0
				    
COPY hello.go /goapp/		    WORKDIR /home/node
WORKDIR "/goapp"		    COPY hello.js hello.js
				    COPY app/package.json app/package.j
RUN go get -d ./...		    RUN (cd app; npm install)
RUN make -f Makefile.goapp	    
				    
FROM scratch			    FROM nabla-node-base
COPY --from=0 /goapp/goapp.seccomp  COPY --from=0 /home/node/app/ /home
ENTRYPOINT ["/goapp.seccomp"]	    CMD ["/home/node/app/app.js"]
```

The only images that we should call "bases" are the ones from which there is some file actually inherited (like the node.seccomp binary). `nabla-go-base` should be renamed `nabla-go-build` or some name that denotes that it's just used for building. The same applies to `C`.

**Comment by useidel** [comment]
Since I am involved in the PR mentioned by @ricarkol above I will take a look at this case since it covers the same "issue" like my C case.
